realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1148,20 +1223,99 @@ public long count() {
      *
      * @return A {@link io.realm.RealmResults} containing objects. If no objects match the condition,
      * a list with zero objects is returned.
-     * @see io.realm.RealmResults
      * @throws java.lang.RuntimeException Any other error
+     * @see io.realm.RealmResults
      */
     public RealmResults<E> findAll() {
         return new RealmResults<E>(realm, query.findAll(), clazz);
     }
 
+
     /**
-     * Find all objects that fulfill the query conditions and sorted by specific field name.
+     * Similar to {@link #findAll()} but runs asynchronously from a worker thread
```

Comment:
Perhaps expand a bit more on the promise contract, e.g. calling `load()` will block. The promise concept will be new to some, so better be detailed about it.